pkg/cloud/services/route53/route53.go

```diff
@@ -313,15 +311,49 @@ func (s *Service) changeManagementClusterDelegation(action string) error {
 }
 
 func (s *Service) createWorkloadClusterZone() error {
+	if s.scope.PrivateZone() && s.scope.VPC() == "" {
+		s.scope.Info("VPC ID is not ready yet for Private Hosted Zone")
+		return aws.ErrMissingEndpoint
```

Comment:
(Not really a blocker, but possible improvement)

If this (VPC still not set, but it will be) is expected state that can happen temporarily, IMHO we should not return an error in that case, but just return and let it retry later. If there are issues in VPC creation, or it takes a bit longer, errors here might create unnecessary noise.

We can set a condition on the CR that describes what is happening, e.g. see these examples in aws-network-topology-operator https://github.com/giantswarm/aws-network-topology-operator/blob/main/controllers/networktopology.go#L105-L114

Reply:
I will drop a ticket as this will need some refactoring of the code so we can return `requeue` if that happens